tests/unit/s2n_client_extensions_test.c

```diff
@@ -118,10 +118,10 @@ static int negotiate_kem(const uint8_t client_extensions[], const size_t client_
     server_conn->kex_params.kem_params.kem = NULL;
 
     /* Send the client hello */
-    POSIX_ENSURE_EQ(write(io_pair->client, record_header, record_header_len), record_header_len);
-    POSIX_ENSURE_EQ(write(io_pair->client, message_header, message_header_len), message_header_len);
-    POSIX_ENSURE_EQ(write(io_pair->client, client_hello_message, client_hello_len), client_hello_len);
-    POSIX_ENSURE_EQ(write(io_pair->client, client_extensions, client_extensions_len), client_extensions_len);
+    POSIX_ENSURE_EQ(write(io_pair->client, record_header, record_header_len), (int64_t) record_header_len);
```

Comment:
It is jarring to go from the generic `ssize_t` and `size_t` types to `int64_t`. I'm not certain if there is a better way to solve this problem. I was going to suggest using a comparison to`SSIZE_MAX` to ensure that we aren't losing information in the cast and then changing the type of `record_header_len` to `ssize_t`. Unfortunately `ssize_t` is non-standard anyway so it's not like that would gain us a bunch of potability.

Reply:
I tried that but the complier complains upon enabling the check. I previously settled for `int`. @camshaft am right in casting to `int64_t` instead`int`.